doc/sphinx/addendum/extraction.rst

```diff
@@ -571,6 +571,25 @@ Additional settings
    This :term:`option` provides a comment that is
    included at the beginning of the output files.
 
+.. opt:: Extraction File Pragma @string
+
+   Specifies a compiler `pragma
+   <https://downloads.haskell.org/ghc/latest/docs/users_guide/exts/pragmas.html#options-ghc-pragma>`_
+   that is included at the beginning of the output files.
+   For example:
+
+   .. coqtop:: in
+
+      Set Extraction File Pragma "OPTIONS_GHC -cpp -XMagicHash".
+
+   The pragma enables C preprocessor and `MagicHash
+   <https://downloads.haskell.org/ghc/latest/docs/users_guide/exts/magic_hash.html#the-magic-hash>`_
+   language extension.
+
+   .. note::
+
+      The option has sense only for extraction to `Haskell or JSON <#grammar-token-language>`_.
+
```

Comment:
```suggestion
   For example, for a Haskell extraction, `Set Extraction File Pragma "OPTIONS_GHC -cpp -XMagicHash".`
   enables the C preprocessor and the `MagicHash
   <https://downloads.haskell.org/ghc/latest/docs/users_guide/exts/magic_hash.html#the-magic-hash>`_
   language extension.
```

Not much benefit in calling coqtop for a single line that doesn't show any output.  And I think easier to read as a single sentence rather than with a lot of formatting.  Just a thought/suggestion.